crates/shim-kvm/src/snp/mod.rs

```diff
@@ -91,3 +92,29 @@ pub fn pvalidate(addr: VirtAddr, size: PvalidateSize, validated: bool) -> Result
         ret => Err(Unknown(ret)),
     }
 }
+
+/// A trait for types that can be serialized and deserialized to/from a byte slice.
+///
+/// # Safety
+///
+/// Behavior is undefined if Self is initialized with bytes, which do not represent a valid state.
+pub unsafe trait ByteSized: Sized {
```

Comment:
Currently this trait is also defined in the Enarx crate, is there a good place for this to live so that all the crates can share it?

Reply:
hmm... sallyport?